### PR TITLE
343 UI Update Admin Tools Scan History table

### LINKED ISSUE #343: UI: Update Admin Tools Scan History Table
# 🐛 Summary #

The Admin Tools  Scan History table needs to be updated to the MUI Data Grid component to stay inline with the rest of the app.

It will also need proper aria-labels and tab indexes for assistive technology users to meet 508 compliance:

- “Details Column “”” buttons inside the Scan History table are not in keyboard tab order. There is no mechanism in which assistive technology users can select the “” button because the interactive element is not keyboard accessible using the either the tab keys or the arrow keys. This happens because the <svg> is coded within a non-interactive <span> element rather than as a <button>.
Add any screenshots of the problem here.

- The “Details” column “” buttons inside the Scan history table do not have accessible name specified for the buttons. Each “” button should have a Unique identifier to distinguish it from the other rows in the table. For example: the aria-label for the first row should be “c488cc67-48d4-4c67-b40b-45d7d5e028a5 Toggle Row Collapsed” (assuming that ID is a unique value.)

## Any helpful log output or screenshots ##

image

### PR DESCRIPTION
**Pull request by hawkishpolicy**
#  #
- Updated ScanTasksView table to MUI Data Grid X
## 🗣 Description ##
- Refactored code to use new Data Grid
- Added aria-labels to Details Icon Button
- Corrected tab indexing of table
- Built new Scan Name dropdown filter
- Built new Scan Status dropdown filter
- Built new modal to house Logs, Inout, Output, and Actions.

## 💭 Motivation and context ##
- Brings Scans History inline with new Organizations table.
- Updates style.
- Improves UX.
- Meets 508 compliance.
- Closes #343 

## 📷 Screenshots (if appropriate) ##
Screenshot 2024-06-25 at 10 37 20 AM
Screenshot 2024-06-25 at 11 04 20 AM
Screenshot 2024-06-25 at 11 04 28 AM

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Review by aloftus23 — CHANGES_REQUESTED** [review]
- [x] Filtering of Name and Status only works if the value exists. Say, dnstwist hasn't been run ever and you filter the Name by dnstwist; the table doesn't change at all. Instead should say No results or something
- [x] The table needs to be sorted by the Created At column where the first entry is the most recent. I believe right now it is doing the opposite

Otherwise looks good!

**Review by aloftus23 — APPROVED** [review]
Looks good!

**Review by schmelz21 — APPROVED** [review]
Looks good...

File: frontend/src/pages/Scans/ScanTasksView.tsx [hunk 1]
```diff
@@ -240,43 +142,368 @@ export const ScanTasksView: React.FC = () => {
...
               page,
+              pageSize: query.pageSize ?? PAGE_SIZE,
               sort: sort[0]?.id ?? 'createdAt',
-              order: sort[0]?.desc ? 'DESC' : 'ASC',
+              order: 'DESC',
               filters: tableFilters
             }
           }
         );
+        // if (result.length === 0) return;
```
Comment (cduhn17) [inline frontend/src/pages/Scans/ScanTasksView.tsx:152]: Is this commented code needed?

**Pull request merged**